lib/rubocop/cop/style/commented_keyword.rb

```diff
@@ -61,25 +47,19 @@ def investigate(processed_source)
         KEYWORDS = %w[begin class def end module].freeze
         ALLOWED_COMMENTS = %w[:nodoc: :yields: rubocop:disable].freeze
 
-        def offensive?(line)
-          line = line.lstrip
-          KEYWORDS.any? { |word| line =~ /^#{word}\s/ } &&
+        def offensive?(comment)
+          line = line(comment)
+          KEYWORDS.any? { |word| line =~ /^\s*#{word}\s/ } &&
             ALLOWED_COMMENTS.none? { |c| line =~ /#\s*#{c}/ }
         end
 
-        def message(node)
-          line = node.source_line
-          keyword = /^\s*(\S+).*#/.match(line)[1]
+        def message(comment)
+          keyword = line(comment)[/(\S+).*#/, 1]
```

Comment:
I think using an explicit `match` is a bit more readable. I've rarely seen the index syntax in the wild and it's likely going to confuse some people.

Reply:
Maybe we need a cop for this … 😜 Changed.

Reply:
I wouldn't mind one. :D In general I think there are very few cases where implementing `[]` results in some more readable code, as for most people it has a very strong association with some indices.